src/main/java/store/controller/ConvenienceController.java

```diff
@@ -0,0 +1,152 @@
+package store.controller;
+
+import store.domain.*;
+import store.dto.AdditionalBonusDto;
+import store.dto.PurchaseItemDto;
+import store.handler.InputHandler;
+import store.service.FileLoadService;
+import store.util.Converter;
+import store.view.InputView;
+import store.view.OutputView;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ConvenienceController {
+
+    private static final int MEMBERSHIP_DISCOUNT_PERCENT = 30;
+    private static final int MAX_MEMBERSHIP_DISCOUNT_AMOUNT = 30;
+
+    private final FileLoadService fileLoadService = new FileLoadService();
+
+    public void run() {
+        Promotions promotions = fileLoadService.loadPromotionFile();
+        TotalProducts totalProducts = fileLoadService.loadProductFile();
+
+        boolean continueShopping = true;
+        while (continueShopping) {
+            OutputView.printWelcomeMessage();
+            OutputView.printProductInfo(totalProducts);
+
+            OutputView.requestProductAndQuantity();
+            String productInfo = InputHandler.receiveValidatedPurchaseInfo();
+            ShoppingCart shoppingCart = addPurchaseItem(productInfo);
+
+            List<AdditionalBonusDto> bonusDtos = new ArrayList<>();
+            processPromotionsForCartItems(shoppingCart, totalProducts, promotions, bonusDtos);
+
+            int membershipDiscount = getMembershipDiscount(shoppingCart, totalProducts);
+
+            // 재고 감소 및 재고 부족 예외 처리
+            if (!attemptToReduceStock(shoppingCart, totalProducts)) continue;
+
+            OutputView.printTotalReceipt(shoppingCart, totalProducts, bonusDtos, membershipDiscount);
+            continueShopping = isContinueShopping();
+        }
```

Comment:
```suggestion
        do {
            OutputView.printWelcomeMessage();
            OutputView.printProductInfo(totalProducts);

            OutputView.requestProductAndQuantity();
            String productInfo = InputHandler.receiveValidatedPurchaseInfo();
            ShoppingCart shoppingCart = addPurchaseItem(productInfo);

            List<AdditionalBonusDto> bonusDtos = new ArrayList<>();
            processPromotionsForCartItems(shoppingCart, totalProducts, promotions, bonusDtos);

            int membershipDiscount = getMembershipDiscount(shoppingCart, totalProducts);

            // 재고 감소 및 재고 부족 예외 처리
            if (!attemptToReduceStock(shoppingCart, totalProducts)) continue;

            OutputView.printTotalReceipt(shoppingCart, totalProducts, bonusDtos, membershipDiscount);
        } while (isContinueShopping())
```
위와 같이 do - while문을 사용하면 추가적인 변수 없이 더 가독성이 좋아진다고 생각하는데 어떤가요?

Reply:
do-while문이 더 깔끔할 것 같습니다! 구현하면서 변수가 거슬렸는데 좋은 방법인 것 같네요! 감사합니다